doc/plutus/explanations/order-book-pattern.rst

```diff
@@ -0,0 +1,119 @@
+
+Generalising the pattern
+------------------------
+
+The original application matches buyers and sellers of currency values, but there are other areas where the pattern is useful.
+
+Off-chain oracles
+~~~~~~~~~~~~~~~~~
+
+Imagine a Plutus script that says "If you give me a the current USD/EUR exchange rate signed by a specific private key, then I will pay you 5 ADA (and use the exchange rate to run the rest of my `$BUSINESS_LOGIC`)".
+The match maker then builds a transaction that combines the oracle value with the Plutus script.
+Of course this example requires the match maker to be able to obtain the signed value, but it does succeed in decoupling consumers of the information from  producers.
+
+On-chain oracles
+~~~~~~~~~~~~~~~~
+
+The pattern could also be a building block for on-chain oracles.
+
+Let's say we have a crypto-backed stablecoin, not too dissimilar to `Djed <https://iohk.io/en/blog/posts/2021/08/18/djed-implementing-algorithmic-stablecoins-for-proven-price-stability/>`_, that relies on recent quotes of the exchange rate between Ada and PEAR tokens.
+And we have a DEX like the one described above where Ada and PEAR are actively traded.
+Every order fulfilled on the DEX gives us a snapshot of the exchange rate at that time.
+Example 1 from above would result in "15.00 ADA / PEAR" (using the buy price here but that's just a technicality).
+
+This is exactly the information that we need for our dealings with the stablecoin, but how do we get it from the DEX to the stablecoin?
+There are two options.
+To choose the right one we need to consider the requirement and usage patterns of our application.
+
+1. Oracle UTXOs
+...............
+
+We could change the DEX contract to produce a new script UTXO for each fulfilled order that records the time and exchange rate of the order.
+The stablecoin user creates a UTXO with a script that requires an oracle UTXO to be present in the spending transaction, and the match maker would put the oracle UTXO and the stablecoing-action UTXO into one transaction and submit it.
+
+2. Oracle tokens
+................
+
+In a variation of the first idea, the DEX could produce *tokens* that encode the script-certified information we are interested in.
+We could set the asset name of the token to be the hash of the exchange rate data, and allow the transaction to produce any quantitiy of these tokens when the order is fulfilled.
+
+The :term:`minting policy <minting policy script>` of the oracle token should allow any number of tokens of the same asset name to be created as long as at least one token with that asset name exists already, reflecting the idea that information is hard to obtain but easy to replicate.
+
+The consumer of the oracle token needs to check that a token with the expected MPS hash is present in the transaction, and that the datum value of the token's asset name is available.
```

Comment:
MPS here means minting-policy script? Maybe it could be a term in the glossary?

Reply:
Yeah I deleted the use of the acronym